<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SettingsDialogPythonConsole</class>
 <widget class="QWidget" name="SettingsDialogPythonConsole">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>809</width>
    <height>974</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>500</width>
    <height>0</height>
   </size>
  </property>
  <property name="font">
   <font>
    <bold>false</bold>
   </font>
  </property>
  <layout class="QGridLayout" name="gridLayout_16">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item row="0" column="0">
    <widget class="QScrollArea" name="scrollArea">
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>-115</y>
        <width>795</width>
        <height>1089</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item row="2" column="0">
        <widget class="QgsCollapsibleGroupBox" name="groupBoxFromatting">
         <property name="title">
          <string>Formatting</string>
         </property>
         <property name="collapsed" stdset="0">
          <bool>false</bool>
         </property>
         <property name="saveCollapsedState" stdset="0">
          <bool>true</bool>
         </property>
         <layout class="QGridLayout" name="gridLayout_4">
          <property name="verticalSpacing">
           <number>2</number>
          </property>
          <item row="1" column="0" colspan="2">
           <widget class="QCheckBox" name="sortImports">
            <property name="text">
             <string>Sort imports</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>Maximum line length</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QgsSpinBox" name="maxLineLength">
            <property name="minimum">
             <number>60</number>
            </property>
            <property name="maximum">
             <number>200</number>
            </property>
            <property name="value">
             <number>80</number>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Formatter</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QgsSpinBox" name="autopep8Level">
            <property name="maximum">
             <number>3</number>
            </property>
            <property name="value">
             <number>1</number>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="autopep8LevelLabel">
            <property name="text">
             <string>Autopep8 level</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0" colspan="2">
           <widget class="QCheckBox" name="blackNormalizeQuotes">
            <property name="text">
             <string>Normalize quotes</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0" colspan="2">
           <widget class="QCheckBox" name="formatOnSave">
            <property name="text">
             <string>Reformat on save</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QComboBox" name="formatter">
            <property name="minimumSize">
             <size>
              <width>100</width>
              <height>0</height>
             </size>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QgsCollapsibleGroupBox" name="groupBoxAutoCompletion">
         <property name="title">
          <string>Autocompletion</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
         <property name="collapsed" stdset="0">
          <bool>false</bool>
         </property>
         <property name="saveCheckedState" stdset="0">
          <bool>true</bool>
         </property>
         <property name="saveCollapsedState" stdset="0">
          <bool>true</bool>
         </property>
         <layout class="QGridLayout" name="gridLayout_5">
          <property name="verticalSpacing">
           <number>2</number>
          </property>
          <item row="0" column="2">
           <widget class="QLabel" name="label_36">
            <property name="text">
             <string>characters</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="autoCompThreshold">
            <property name="maximum">
             <number>10</number>
            </property>
            <property name="value">
             <number>2</number>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Autocompletion threshold</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="3">
           <layout class="QGridLayout" name="layoutRadioButton">
            <item row="0" column="2">
             <widget class="QRadioButton" name="autoCompFromDocAPI">
              <property name="toolTip">
               <string>Get autocompletion from current document and installed APIs</string>
              </property>
              <property name="text">
               <string>From doc and APIs</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QRadioButton" name="autoCompFromAPI">
              <property name="toolTip">
               <string>Get autocompletion from installed APIs</string>
              </property>
              <property name="text">
               <string>From API files</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QRadioButton" name="autoCompFromDoc">
              <property name="toolTip">
               <string>Get autocompletion from current document</string>
              </property>
              <property name="text">
               <string>From document</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QgsCollapsibleGroupBox" name="groupBoxApi">
         <property name="title">
          <string>APIs</string>
         </property>
         <property name="collapsed" stdset="0">
          <bool>true</bool>
         </property>
         <property name="saveCollapsedState" stdset="0">
          <bool>true</bool>
         </property>
         <layout class="QGridLayout" name="gridLayout_7">
          <item row="0" column="0">
           <widget class="QCheckBox" name="preloadAPI">
            <property name="text">
             <string>Using preloaded APIs file</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <layout class="QGridLayout" name="gridLayout_2">
            <item row="0" column="1">
             <layout class="QVBoxLayout" name="verticalLayout">
              <item>
               <widget class="QPushButton" name="addAPIpath">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="minimumSize">
                 <size>
                  <width>32</width>
                  <height>32</height>
                 </size>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>32</width>
                  <height>32</height>
                 </size>
                </property>
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="removeAPIpath">
                <property name="minimumSize">
                 <size>
                  <width>32</width>
                  <height>32</height>
                 </size>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>32</width>
                  <height>32</height>
                 </size>
                </property>
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="verticalSpacer">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
            <item row="0" column="0">
             <widget class="QTableWidget" name="tableWidget">
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>100</height>
               </size>
              </property>
              <property name="editTriggers">
               <set>QAbstractItemView::NoEditTriggers</set>
              </property>
              <property name="tabKeyNavigation">
               <bool>false</bool>
              </property>
              <property name="showDropIndicator" stdset="0">
               <bool>false</bool>
              </property>
              <property name="dragDropOverwriteMode">
               <bool>false</bool>
              </property>
              <property name="alternatingRowColors">
               <bool>true</bool>
              </property>
              <property name="selectionMode">
               <enum>QAbstractItemView::ExtendedSelection</enum>
              </property>
              <property name="selectionBehavior">
               <enum>QAbstractItemView::SelectRows</enum>
              </property>
              <property name="verticalScrollMode">
               <enum>QAbstractItemView::ScrollPerPixel</enum>
              </property>
              <property name="horizontalScrollMode">
               <enum>QAbstractItemView::ScrollPerPixel</enum>
              </property>
              <property name="gridStyle">
               <enum>Qt::SolidLine</enum>
              </property>
              <property name="rowCount">
               <number>0</number>
              </property>
              <property name="columnCount">
               <number>2</number>
              </property>
              <attribute name="horizontalHeaderVisible">
               <bool>false</bool>
              </attribute>
              <attribute name="horizontalHeaderCascadingSectionResizes">
               <bool>false</bool>
              </attribute>
              <attribute name="horizontalHeaderShowSortIndicator" stdset="0">
               <bool>false</bool>
              </attribute>
              <attribute name="horizontalHeaderStretchLastSection">
               <bool>true</bool>
              </attribute>
              <attribute name="verticalHeaderVisible">
               <bool>false</bool>
              </attribute>
              <column>
               <property name="text">
                <string>APIs</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Path</string>
               </property>
              </column>
             </widget>
            </item>
           </layout>
          </item>
          <item row="2" column="0">
           <widget class="QgsCollapsibleGroupBox" name="groupBoxPreparedAPI">
            <property name="title">
             <string>Using prepared APIs file</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <property name="checked">
             <bool>false</bool>
            </property>
            <property name="collapsed" stdset="0">
             <bool>true</bool>
            </property>
            <property name="saveCollapsedState" stdset="0">
             <bool>true</bool>
            </property>
            <property name="saveCheckedState" stdset="0">
             <bool>true</bool>
            </property>
            <layout class="QGridLayout" name="gridLayout_3">
             <item row="0" column="0">
              <widget class="QPushButton" name="compileAPIs">
               <property name="text">
                <string>Compile APIs…</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLineEdit" name="lineEdit">
               <property name="minimumSize">
                <size>
                 <width>0</width>
                 <height>27</height>
                </size>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QgsCollapsibleGroupBox" name="groupBoxRunDebugEditor">
         <property name="title">
          <string>Run and Debug</string>
         </property>
         <property name="collapsed" stdset="0">
          <bool>true</bool>
         </property>
         <property name="saveCollapsedState" stdset="0">
          <bool>true</bool>
         </property>
         <layout class="QGridLayout" name="gridLayout_21">
          <property name="verticalSpacing">
           <number>2</number>
          </property>
          <item row="1" column="0">
           <widget class="QCheckBox" name="autoSaveScript">
            <property name="text">
             <string>Auto-save script before running</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QCheckBox" name="enableObjectInspector">
            <property name="text">
             <string>Enable Object Inspector (switching between tabs may be slow)</string>
            </property>
            <property name="checked">
             <bool>false</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="7" column="0">
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="1" column="0">
        <widget class="QgsCollapsibleGroupBox" name="groupBoxTyping">
         <property name="title">
          <string>Typing</string>
         </property>
         <property name="collapsed" stdset="0">
          <bool>false</bool>
         </property>
         <property name="saveCollapsedState" stdset="0">
          <bool>true</bool>
         </property>
         <layout class="QGridLayout" name="gridLayout_11">
          <property name="verticalSpacing">
           <number>2</number>
          </property>
          <item row="0" column="0">
           <widget class="QCheckBox" name="autoCloseBracket">
            <property name="text">
             <string>Automatic parentheses insertion</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QCheckBox" name="autoSurround">
            <property name="text">
             <string>Automatically surround selection when typing quotes or brackets</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QCheckBox" name="autoInsertImport">
            <property name="text">
             <string>Automatic insertion of the 'import' string on 'from xxx'</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="6" column="0">
        <widget class="QgsCollapsibleGroupBox" name="groupBox_2">
         <property name="title">
          <string>External Editor</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_8">
          <item row="0" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Command to launch an external Python code editor. If empty, the default system editor will be used.&lt;/p&gt;&lt;p&gt;Use the token &lt;span style=&quot; font-style:italic;&quot;&gt;&amp;lt;file&amp;gt;&lt;/span&gt; to insert the filename, &lt;span style=&quot; font-style:italic;&quot;&gt;&amp;lt;line&amp;gt;&lt;/span&gt; to insert line number, and &lt;span style=&quot; font-style:italic;&quot;&gt;&amp;lt;col&amp;gt;&lt;/span&gt; to insert the column number. For example:&lt;br/&gt;&lt;span style=&quot; font-family:'Noto Sans Mono';&quot;&gt;kate -l &amp;lt;line&amp;gt; -c &amp;lt;col&amp;gt; &amp;quot;&amp;lt;file&amp;gt;&amp;quot;&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
            <property name="textInteractionFlags">
             <set>Qt::TextBrowserInteraction</set>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QgsFilterLineEdit" name="externalEditor">
            <property name="placeholderText">
             <string>Default</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QgsCollapsibleGroupBox</class>
   <extends>QGroupBox</extends>
   <header>qgis.gui</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsSpinBox</class>
   <extends>QSpinBox</extends>
   <header>qgis.gui</header>
  </customwidget>
  <customwidget>
   <class>QgsFilterLineEdit</class>
   <extends>QLineEdit</extends>
   <header>qgis.gui</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>groupBoxAutoCompletion</tabstop>
  <tabstop>scrollArea</tabstop>
  <tabstop>autoCompThreshold</tabstop>
  <tabstop>autoCompFromDoc</tabstop>
  <tabstop>autoCompFromAPI</tabstop>
  <tabstop>autoCompFromDocAPI</tabstop>
  <tabstop>autoCloseBracket</tabstop>
  <tabstop>autoSurround</tabstop>
  <tabstop>autoInsertImport</tabstop>
  <tabstop>formatOnSave</tabstop>
  <tabstop>sortImports</tabstop>
  <tabstop>maxLineLength</tabstop>
  <tabstop>formatter</tabstop>
  <tabstop>autopep8Level</tabstop>
  <tabstop>blackNormalizeQuotes</tabstop>
  <tabstop>enableObjectInspector</tabstop>
  <tabstop>autoSaveScript</tabstop>
  <tabstop>preloadAPI</tabstop>
  <tabstop>tableWidget</tabstop>
  <tabstop>addAPIpath</tabstop>
  <tabstop>removeAPIpath</tabstop>
  <tabstop>groupBoxPreparedAPI</tabstop>
  <tabstop>compileAPIs</tabstop>
  <tabstop>lineEdit</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
